R/list2df_linter.R

```diff
@@ -0,0 +1,37 @@
+#' Recommend direct usage of `data.frame()` to create a data.frame from a list
+#'
+#' It is possible to create a data.frame from a list of columns with `data.frame()`
+#' or `list2DF()` (since \R 4.0.0 and if recycling is not required), rather than
+#' iteratively adding columns with `cbind()`.
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "do.call(cbind.data.frame, x)",
+#'   linters = list2df_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "list2df(x)",
+#'   linters = list2df_linter()
+#' )
+#'
+#' lint(
+#'   text = "data.frame(list(letters, LETTERS))",
+#'   linters = list2df_linter()
+#' )
+#'
+#' @evalRd rd_tags("list2df_linter")
+#' @seealso [linters] for a complete list of linters available in lintr.
+#' @export
+list2df_linter <- make_linter_from_function_xpath(
+  function_names = "do.call",
+  xpath = "./following-sibling::expr[1]/SYMBOL[text() = 'cbind.data.frame']",
```

Comment:
it should be easy to catch `do.call("cbind.data.frame"` too:

https://github.com/search?q=lang%3AR%20%2Fdo%5B.%5Dcall%2F%20%2F%22cbind%5B.%5Ddata%5B.%5Dframe%22%2F&type=code

Though I think it will require a "more traditional" linter structure than `make_linter_from_function_xpath()` since we'll need `get_r_string()` to reliable compute the `STR_CONST` value.